capstone/src/main/webapp/scripts/profile.js

```diff
@@ -0,0 +1,124 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/*
+ * Initializes the page.
+ */
+ window.addEventListener('load', loadPage);
+
+/*
+ * Handles all functions to trigger when page loads.
+ */
+function loadPage() {
+  getUserData();
+}
+
+/** Fetch current user's data from the server */
+function getUserData() {
+  fetch('/user')
+  .then(response => response.json())
+  .then((user) => {
+    displayUserInfo(user);
+  });
+}
+
+/** Display current user's information on the page */
+function displayUserInfo(user) {
+  nameContainer = document.getElementById('name-container');
+  nameContainer.innerHTML = user.name;
+
+  emailContainer = document.getElementById('email-container');
+  emailContainer.innerHTML = user.email;
+
+  phoneContainer = document.getElementById('phone-container');
+  phoneContainer.innerHTML = user.phoneNumber;
+
+  displayInterests(user.interests);
+  displayProfilePicture(user.profilePic);
+
+  getGroupData(user.groups);
+  
+  displayBadges(user.badges); // maybe change this to getBadgeData?
```

Comment:
I think calling it displayBadges is fine. What do you mean add it to Badge and User class?

Reply:
Oh oops, those were comments to myself I forgot to remove. I think similar to how I have a separate getGroupData() function that gets data from the servlet, we might need to do the same thing with Badges. In that case, would have to call getBadgeData() before displayBadges(). Another option, which I think I like better, would be where I would just have the Badge data returned with the User object. Right now I'm not turning the list of badgeIds from database into a list of Badge objects - that's what I meant by adding the functionality to the Badge and User classes. It's currently a TODO in my User fromEntity() method